core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -106,19 +113,38 @@ oppia.directive('questionsList', [
 
           ctrl.goToNextPage = function() {
             ctrl.currentPage++;
-            var questionSummaries =
-              ctrl.getQuestionSummaries(ctrl.currentPage);
-            if (questionSummaries === null) {
-              ctrl.fetchQuestionSummaries(ctrl.entityId, false);
-            } else {
-              ctrl.questionSummaries = questionSummaries;
+            ctrl.questionSummaries = ctrl.getQuestionSummariesAsync(
+              ctrl.currentPage, ctrl.skillIdsList, true, false
+            );
+            if (ctrl.questionSummaries) {
+              ctrl.truncatedQuestionSummaries =
+                ctrl.questionSummaries.map(function(question) {
+                  var summary = $filter(
+                    'formatRtePreview')(question.summary.question_content);
+                  summary = $filter('truncate')(summary, 100);
+                  return summary;
+                });
             }
           };
 
           ctrl.goToPreviousPage = function() {
             ctrl.currentPage--;
-            ctrl.questionSummaries =
-              ctrl.getQuestionSummaries(ctrl.currentPage);
+            ctrl.questionSummaries = ctrl.getQuestionSummariesAsync(
+              ctrl.currentPage, ctrl.skillIdsList, false, false
+            );
+            if (ctrl.questionSummaries) {
+              ctrl.truncatedQuestionSummaries =
```

Comment:
Do not repeat code. Find a way to reduce repetition, here and elsewhere.

Reply:
Done.